src/main/scala/com/wavesplatform/transaction/smart/script/v1/ScriptV1.scala

```diff
@@ -38,6 +38,26 @@ object ScriptV1 {
         val s = Array(version.toByte) ++ Serde.serialize(expr)
         ByteStr(s ++ crypto.secureHash(s).take(checksumLength))
       }
+    override val maxBlockVersion: Coeval[Int] = Coeval.evalOnce(calcMaxBlockVersion(expr))
+  }
+
+  def calcMaxBlockVersion(e: EXPR): Int = {
+    def horTraversal(queue: Iterable[EXPR]): Int = {
```

Comment:
We need to consider very deep expressions too.
Please finc `stack overflow check` tests and implement something similar